app/recipeapp.js

```diff
@@ -293,3 +233,44 @@ function updateTime() {
 }
 
 setInterval(updateTime, 1000);
+
+function fetchIngredientPrice(ingredient) {
+  console.log(typeof ingredient);
+  const keys = Object.keys(ingredientsPrices);
+
+  if (keys.includes(ingredient)) {
+    return `<p>Price of ${ingredient}: $${ingredientsPrices[ingredient]}</p>`;
+  } else {
+    return `<p>${ingredient} not found</p>`;
+  }
+}
+
+const ingredientSearchButton = document.getElementById(
+  "ingredient-search-button"
+);
+if (ingredientSearchButton) {
+  ingredientSearchButton.addEventListener("click", async () => {
+    const ingredientInput = document.getElementById("ingredient-search-input");
+    const resultsDiv = document.getElementById("ingredient-price-result");
+    if (!ingredientInput || !resultsDiv) {
+      console.error("Ingredient input or result div not found.");
+      return;
+    }
+
+    const ingredient = ingredientInput.value.toLowerCase().trim();
+    if (!ingredient) {
+      resultsDiv.innerHTML = "<p>Please enter an ingredient.</p>";
+      return;
+    }
+
+    try {
+      const result = await fetchIngredientPrice(ingredient);
```

Comment:
The `fetchIngredientPrice` function isn't really asynchronous in your case, so having `await` here is completely optional. JavaScript promises are actually quite flexible so both ways work fine though. 

However, just to improve readability a bit we can do:

```suggestion
      const result = getIngredientPrice(ingredient);
```

I used `get*` prefix here in the function name intentionally, since `fetchSomething` is a standard convention for naming function that loads data asynchronously.